src/StdLib/StdDeque.jl

```diff
@@ -4,8 +4,14 @@ Base.IndexStyle(::Type{<:StdDeque}) = IndexLinear()
 Base.size(v::StdDeque) = (Int(cppsize(v)),)
 Base.getindex(v::StdDeque, i::Int) = cxxgetindex(v, i)[]
 Base.setindex!(v::StdDeque{T}, val, i::Int) where {T} = cxxsetindex!(v, convert(T, val), i)
-Base.push!(v::StdDeque, x) = push_back!(v, x)
 Base.pushfirst!(v::StdDeque, x) = push_front!(v, x)
+Base.push!(v::StdDeque, x) = isempty(v) ? push_front!(v, x) : push_back!(v, x)
```

Comment:
The push methods for `StdDeque` also need to return the container, as is done for the other types.

Reply:
Fixed